src/NServiceBus.Gateway.AcceptanceTests/When_listening_on_multiple_uris.cs

```diff
@@ -0,0 +1,137 @@
+
+            while (true)
+            {
+                try
+                {
+                    using (var myWebResponse = (HttpWebResponse)webRequest.GetResponse())
+                    {
+                        if (myWebResponse.StatusCode == HttpStatusCode.OK)
+                        {
+                            break;
+                        }
+                    }
+                }
+                catch (WebException)
+                {
+                }
+            }
+        }
+
+        static HttpWebRequest CreateWebRequest(string uri)
+        {
+#pragma warning disable DE0003 // API is deprecated
+            var webRequest = (HttpWebRequest)WebRequest.Create(uri);
+#pragma warning restore DE0003 // API is deprecated
+            webRequest.Method = "POST";
+            webRequest.ContentType = "text/xml; charset=utf-8";
+            webRequest.UserAgent = "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)";
+
+            webRequest.Headers.Add("Content-Encoding", "utf-8");
+            webRequest.Headers.Add("NServiceBus.CallType", "SingleCallSubmit");
+            webRequest.Headers.Add("NServiceBus.AutoAck", "true");
+            webRequest.Headers.Add("SentToHeader", uri);
+            webRequest.Headers.Add("NServiceBus.Id", Guid.NewGuid().ToString("N"));
+
+            const string message = "<?xml version=\"1.0\" ?><Messages xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" xmlns=\"http://tempuri.net/NServiceBus.Gateway.AcceptanceTests\"><MyRequest></MyRequest></Messages>";
+
+            using (var messagePayload = new MemoryStream(Encoding.UTF8.GetBytes(message)))
+            {
+                webRequest.Headers.Add(HttpRequestHeader.ContentMd5, HttpUtility.UrlEncode(Hash(messagePayload)));
+                webRequest.ContentLength = messagePayload.Length;
+
+                using (var requestStream = webRequest.GetRequestStream())
+                {
+                    messagePayload.CopyTo(requestStream);
+                }
+            }
+
+            return webRequest;
+        }
+
+        static string Hash(Stream stream)
+        {
+            var position = stream.Position;
+            var hash = MD5.Create().ComputeHash(stream);
+
+            stream.Position = position;
+
+            return Convert.ToBase64String(hash);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool GotMessageOnDefaultChannel { get; set; }
+            public bool GotMessageOnNonDefaultChannel { get; set; }
+        }
+
+        public class Headquarters : EndpointConfigurationBuilder
+        {
+            public Headquarters()
+            {
+                EndpointSetup<GatewayEndpoint>(c =>
+                {
+                    c.Gateway().AddReceiveChannel(DefaultReceiveURI);
+                    c.Gateway().AddReceiveChannel($"http://{Dns.GetHostName()}:25898/Headquarters/");
```

Comment:
have you disabled this line and checked whether the test actually fails?

Reply:
Yes it does fail